From e51be84b6dec0e1eadd3c7910fd0e414951a40e6 Mon Sep 17 00:00:00 2001
From: IotaBread <iotabread@gmail.com>
Date: Thu, 17 Aug 2023 21:12:37 -0400
Subject: [PATCH 1/2] Fix git clone

---
 setup.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.ts b/setup.ts
index eb6cf02..bb19373 100644
--- a/setup.ts
+++ b/setup.ts
@@ -21,7 +21,7 @@ for (const dirEntry of Deno.readDirSync("./")) {
 
 // clone
 console.log('Cloning old qm version...')
-await exec('git clone https://github.com/quiltmc/quilt-mappings.git ' + from_version);
+await exec('git clone https://github.com/quiltmc/quilt-mappings.git --depth 1 --branch ' + from_version + ' --single-branch ' + from_version);
 
 console.log('Copying to new version...')
 await exec('cp -r ' + from_version + ' ' + to_version);
-- 
2.34.1

